process_report/process_report.py

```diff
@@ -1,9 +1,106 @@
 import argparse
 import os
+import sys
 
 import pandas
 
 
+### Invoice field names
+PROJECT_FIELD = 'Project - Allocation'
+MANAGER_PI_FIELD = 'Manager (PI)'
+INSTITUTION_FIELD = 'Institution'
+COST_FIELD = 'Cost'
+CREDIT_FIELD = 'Credit'
+CREDIT_CODE_FIELD = 'Credit Code'
+BALANCE_FIELD = 'Balance'
+###
+
+
+def apply_credits_0001(dataframe):
+    credit_code = "0001"
+    pass
+
+
+def apply_credits_0002(dataframe):
+    """Applies the New PI Credit. This credit function expects the 
+    env var `C0002_OLD_PI` to be set, pointing to a txt file containing old PIs"""
+    credit_code = "0002"
+    credit_amount = 1000
+
+    old_pi_list = set()
+    try:
+        with open(os.getenv("C0002_OLD_PI")) as f:
+            for pi in f: old_pi_list.add(pi.strip())
+
+    except Exception:
+        print("Old PI file does not exist, or the C0002_OLD_PI env var is not set")
+        sys.exit(1)
+
+    print("Old pi list: ", old_pi_list)
+    pi_list = dataframe[MANAGER_PI_FIELD].unique()
+
+    for pi in pi_list:
+        if pi != pi: continue   # NaN check
+        if pi in old_pi_list: continue  # Is the PI an old PI?
```

Comment:
If this code is run for testing will the count of months the PI has been on the old PI list go up or are we actually tracking months?

Reply:
@joachimweyl My code has no logic to track how long a PI has appeared on an invoice.

Reply:
see discussion in unit_tests.py line 207